src/main/java/au/org/emii/netcdf/iterator/reader/NetcdfReader.java

```diff
@@ -11,15 +11,19 @@
 import ucar.ma2.ArrayChar;
 import ucar.ma2.Index;
 import ucar.nc2.Variable;
+import ucar.nc2.dataset.EnhanceScaleMissing;
 
 public class NetcdfReader {
 
     private Array data;
     
     private IndexValuesConverter indexValuesConverter;
     private Index index;
+    private final Variable variable;
 
     public NetcdfReader(Variable variable) throws IOException {
```

Comment:
Not sure I have a problem with that in this case. Not doing this will complicate the client api or this class without refactoring which is probably best left to https://github.com/aodn/netcdf-iterator/issues/2.